extensions/elytron-security/runtime/src/main/java/io/quarkus/elytron/security/runtime/SecurityRecorder.java

```diff
@@ -155,6 +165,50 @@ public void loadRealm(RuntimeValue<SecurityRealm> realm, MPRealmConfig config) t
         return new RuntimeValue<>(realm);
     }
 
+    /**
+     * Create a runtime value for a {@linkplain JdbcSecurityRealm}
+     *
+     * @param config - the realm config
+     * @return - runtime value wrapper for the SecurityRealm
+     */
+    public RuntimeValue<SecurityRealm> createRealm(JdbcSecurityRealmConfig config) {
+        log.debugf("createRealm, config=%s", config);
+        Supplier<Provider[]> providers = new Supplier<Provider[]>() {
+            @Override
+            public Provider[] get() {
+                return new Provider[] { new WildFlyElytronProvider() };
+            }
+        };
+        Security.addProvider(new WildFlyElytronProvider());
+        JdbcSecurityRealmBuilder builder = JdbcSecurityRealm.builder().setProviders(providers);
+        PrincipalQueriesConfig principalQueries = config.principalQueries;
+        registerPrincipalQuery(principalQueries.defaultPrincipalQuery, builder);
+        principalQueries.namedPrincipalQueries
+                .forEach((name, principalQuery) -> registerPrincipalQuery(principalQuery, builder));
+        return new RuntimeValue<>(builder.build());
+    }
+
+    private void registerPrincipalQuery(PrincipalQueryConfig principalQuery, JdbcSecurityRealmBuilder builder) {
+        DataSource dataSource = (DataSource) principalQuery.datasource
+                .map(name -> Arc.container().instance(name).get())
+                .orElse(Arc.container().instance(DataSource.class).get());
+
+        QueryBuilder queryBuilder = builder.principalQuery(principalQuery.sql).from(dataSource);
+
+        AttributeMapper[] mappers = principalQuery.attributeMappings.entrySet()
```

Comment:
This seems over complicated. The only attribute that we need is the group one no?
Or do we enable a developer to load additional user information in the table and add them as attributes ?